src/amo/pages/AddonInfo/index.js

```diff
@@ -144,6 +146,24 @@ export class AddonInfoBase extends React.Component<InternalProps> {
       infoType,
     } = this.props;
 
+    if (errorHandler.hasError()) {
+      log.warn(`Captured API Error: ${errorHandler.capturedError.messages}`);
+
+      // 401 and 403 for an add-on lookup is made to look like a 404 on purpose.
+      // See https://github.com/mozilla/addons-frontend/issues/3061
+      if (
+        errorHandler.capturedError.responseStatusCode === 401 ||
+        errorHandler.capturedError.responseStatusCode === 403 ||
+        errorHandler.capturedError.responseStatusCode === 404
+      ) {
+        return <NotFoundPage />;
+      }
+
+      if (errorHandler.capturedError.responseStatusCode === 451) {
+        return <UnavailableForLegalReasonsPage />;
+      }
+    }
+
```

Comment:
this seems to be the same code as in Addon/index.js above - DRY?

Reply:
and in some further components too 😿

Reply:
It's tricky to refactor here.

Reply:
Some refactoring is likely possible, but then it becomes more opaque, so it's a bit of a tradeoff. This is generally the way the code is written for pages, with everything that happens called out pretty specifically, so perhaps we should just keep it this way for now.

@willdurand Do you think it's worth extracting this out into a function which is used by every page that needs it, or are you suggesting we just stick with how it is now?

Reply:
> @willdurand Do you think it's worth extracting this out into a function which is used by every page that needs it, or are you suggesting we just stick with how it is now?

I'd say no.

Reply:
We could file an issue to investigate a better strategy since we have this `Page` component now but no need to change anything here.